File: src/16.ts
import _ from 'lodash';

// Computes the sum of the digits of the given number raised to the given power.
//
// See {@link https://projecteuler.net/problem=16}.
export default function compute(k: number, e: number) {
  const s = BigInt(Math.pow(k, e)).toString();
  return s.split('').reduce((a, b) => a + _.toNumber(b), 0);
}

File: project_euler/16.py
# Problem 16: Power Digit Sum
#
# What is the sum of the digits of the number 2^1000?
#
# See https://projecteuler.net/problem=16
def run():
    s = str(2**1000)
    return sum(int(c) for c in s)


def test_run():
    assert run() == 1366
